core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -463,4 +475,171 @@ public void testRecursiveTimeFilterRegex() throws Exception {
+  }
+
+  @Test
+  public void testFileBatchInputFormatAvro() throws Exception {
+    Schema recordSchema = Schema.recordOf("record",
+                                          Schema.Field.of("i", Schema.of(Schema.Type.INT)),
+                                          Schema.Field.of("l", Schema.of(Schema.Type.LONG)));
+
+    String filePath = "file:///tmp/test/test.avro";
+    Path avroFilePath = new Path(filePath);
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+
+    ETLStage source = new ETLStage(
+      "source", new ETLPlugin("File", BatchSource.PLUGIN_TYPE,
+                                    ImmutableMap.<String, String>builder()
+                                      .put(Constants.Reference.REFERENCE_NAME, "TestFile")
+                                      .put(Properties.File.FILESYSTEM, "Text")
+                                      .put(Properties.File.PATH, filePath)
+                                      .put(Properties.File.FILE_SOURCE_FORMAT, "Avro")
+                                      .put(Properties.File.IGNORE_NON_EXISTING_FOLDERS, "false")
+                                      .build(),
+                                      null));
+
+    String outputDatasetName = "test-filesource-avro";
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(outputDatasetName));
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(sink)
+      .addConnection(source.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(BATCH_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("FileSourceAvroFormat");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(recordSchema.toString());
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("i", Integer.MAX_VALUE)
+      .set("l", Long.MAX_VALUE)
+      .build();
+
+    DataSetManager<TimePartitionedFileSet> inputManager = getDataset("TestFile");
+
+    DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(avroSchema);
+    DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+    dataFileWriter.create(avroSchema, fs.create(avroFilePath));
+    dataFileWriter.append(record);
+    dataFileWriter.close();
+    inputManager.flush();
+
+    WorkflowManager workflowManager = appManager.getWorkflowManager(SmartWorkflow.NAME);
+    workflowManager.start();
+    workflowManager.waitForRuns(ProgramRunStatus.COMPLETED, 1, 2, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
+    List<StructuredRecord> output = MockSink.readOutput(outputManager);
+    Assert.assertEquals(1, output.size());
+    Assert.assertEquals(Integer.MAX_VALUE, output.get(0).get("i"));
```

Comment:
as mentioned before, this is an incomplete comparison. Fields are missing and schema is wrong, but the test is not catching that. It should be comparing object directly instead of a couple fields.